params/config.go.orig

```diff
@@ -0,0 +1,577 @@
+	}
+
+	// AllGxhashProtocolChanges contains every protocol change (GxIPs) introduced
+	// and accepted by the klaytn developers into the Klaytn consensus.
+	//
+	// This configuration is intentionally not using keyed fields to force anyone
+	// adding flags to the config to also have to set these fields.
+	AllGxhashProtocolChanges = &ChainConfig{
+		ChainID:  big.NewInt(0),
+		Gxhash:   new(GxhashConfig),
+		Clique:   nil,
+		Istanbul: nil,
+	}
+
+	// AllCliqueProtocolChanges contains every protocol change (GxIPs) introduced
+	// and accepted by the klaytn developers into the Clique consensus.
+	//
+	// This configuration is intentionally not using keyed fields to force anyone
+	// adding flags to the config to also have to set these fields.
+	AllCliqueProtocolChanges = &ChainConfig{
+		ChainID:  big.NewInt(0),
+		Gxhash:   nil,
+		Clique:   &CliqueConfig{Period: 0, Epoch: 30000},
+		Istanbul: nil,
+	}
+
+	TestChainConfig = &ChainConfig{
+		ChainID:       big.NewInt(1),
+		Gxhash:        new(GxhashConfig),
+		Clique:        nil,
+		Istanbul:      nil,
+		UnitPrice:     1, // NOTE-Klaytn Use UnitPrice 1 for tests
+		DeriveShaImpl: 0,
+	}
+	TestRules = TestChainConfig.Rules(new(big.Int))
+
+	// istanbul BFT
+	BFTTestChainConfig = &ChainConfig{
+		ChainID:  big.NewInt(1),
+		Gxhash:   new(GxhashConfig),
+		Clique:   nil,
+		Istanbul: nil,
+	}
+)
+
+// VMLogTarget sets the output target of vmlog.
+// The values below can be OR'ed.
+//  - 0x0: no output (default)
+//  - 0x1: file (DATADIR/logs/vm.log)
+//  - 0x2: stdout (like logger.DEBUG)
+var VMLogTarget = 0x0
+
+const (
+	VMLogToFile   = 0x1
+	VMLogToStdout = 0x2
+	VMLogToAll    = VMLogToFile | VMLogToStdout
+
+	UpperGasLimit = uint64(999999999999)
+)
+
+const (
+	PasswordLength = 16
+)
+
+// ChainConfig is the blockchain config which determines the blockchain settings.
+//
+// ChainConfig is stored in the database on a per block basis. This means
+// that any network, identified by its genesis block, can have its own
+// set of configuration options.
+type ChainConfig struct {
+	ChainID *big.Int `json:"chainId"` // chainId identifies the current chain and is used for replay protection
+
+	// "Compatible" means that it is EVM compatible(the opcode and precompiled contracts are the same as Ethereum EVM).
+	// In other words, not all the hard fork items are included.
+	IstanbulCompatibleBlock  *big.Int `json:"istanbulCompatibleBlock,omitempty"`  // IstanbulCompatibleBlock switch block (nil = no fork, 0 = already on istanbul)
+	LondonCompatibleBlock    *big.Int `json:"londonCompatibleBlock,omitempty"`    // LondonCompatibleBlock switch block (nil = no fork, 0 = already on london)
+	EthTxTypeCompatibleBlock *big.Int `json:"ethTxTypeCompatibleBlock,omitempty"` // EthTxTypeCompatibleBlock switch block (nil = no fork, 0 = already on ethTxType)
+	MagmaCompatibleBlock     *big.Int `json:"magmaCompatibleBlock,omitempty"`     // MagmaCompatible switch block (nil = no fork, 0 already on Magma)
+<<<<<<< HEAD
```

Comment:
Do we need to preserve this kinds of conflict statements?

Reply:
Wrong file uploaded. sorry for the hassle